File: java/src/main/java/org/dilithia/sdk/model/DeployPayload.java
package org.dilithia.sdk.model;

/**
 * Represents a contract deployment or upgrade payload.
 *
 * @param name      the contract name
 * @param bytecode  the hex-encoded bytecode
 * @param from      the sender address
 * @param alg       the signing algorithm
 * @param pk        the public key
 * @param sig       the signature in hex
 * @param nonce     the sender's nonce
 * @param chainId   the chain identifier
 * @param version   the contract version
 */
public record DeployPayload(
        String name,
        String bytecode,
        Address from,
        String alg,
        PublicKey pk,
        String sig,
        long nonce,
        String chainId,
        int version
) {}

File: dotnet/Dilithia.Sdk/Models/DeployPayload.cs
namespace Dilithia.Sdk.Models;

/// <summary>
/// Payload for contract deploy/upgrade requests.
/// </summary>
/// <param name="Name">Contract name.</param>
/// <param name="Bytecode">Hex-encoded contract bytecode.</param>
/// <param name="From">Deployer address.</param>
/// <param name="Alg">Signature algorithm identifier.</param>
/// <param name="Pk">Public key (hex).</param>
/// <param name="Sig">Signature (hex).</param>
/// <param name="Nonce">Account nonce.</param>
/// <param name="ChainId">Target chain identifier.</param>
/// <param name="Version">Optional contract version for upgrades.</param>
public record DeployPayload(
    string Name,
    string Bytecode,
    string From,
    string Alg,
    string Pk,
    string Sig,
    long Nonce,
    string ChainId,
    int? Version = null
);
